src/redux/selectors/index.js

```diff
@@ -357,6 +359,78 @@ export const collectionFlattenedSubTypesSelector = createSelector(
   },
 );
 
+export const previewCollectionResourceIdsSelector = createSelector(
+  [collectionsSelector, previewCollectionIdSelector],
+  (collections, previewCollectionId) => collections[previewCollectionId]?.resourceIds,
+);
+
+const previewFilteredResourceTypesSelector = createSelector(
+  [
+    previewCollectionResourceIdsSelector,
+    timelineItemsInRangeSelector,
+    subTypeResourceIdsSelector,
+  ],
+  (
+    previewCollectionResourceIdsObjects,
+    timelineItemsInRange,
+    subTypeResourceIds,
+  ) => {
+    if (!previewCollectionResourceIdsObjects) {
+      return {};
+    }
+    const selectedCollectionResourceIds = Object.keys(previewCollectionResourceIdsObjects);
+    return [...timelineItemsInRange].reverse().reduce((acc, { id, type, subType }) => {
+      if (!acc[type]) {
+        acc[type] = {};
+      }
+      if (!acc[type].subTypes) {
+        acc[type].subTypes = {};
+      }
+      if (!acc[type].subTypes[subType]) {
+        acc[type].subTypes[subType] = {};
+        acc[type].subTypes[subType].resourceIds = Array.from(subTypeResourceIds[subType]);
+        acc[type].subTypes[subType].count = Array.from(subTypeResourceIds[subType]).length;
+        acc[type].subTypes[subType].dateFilteredResourceIds = [];
+        acc[type].subTypes[subType].dateFilteredCount = 0;
+        acc[type].subTypes[subType].collectionDateFilteredResourceIds = [];
+        acc[type].subTypes[subType].collectionDateFilteredCount = 0;
+      }
+      acc[type].subTypes[subType].dateFilteredResourceIds.push(id);
+      acc[type].subTypes[subType].dateFilteredCount = (
+        acc[type].subTypes[subType].dateFilteredResourceIds.length
+      );
```

Comment:
Is there any possible re-use vis-a-vis the existing `filteredResourceTypesSelector`  ?
(if the shape is similar)

Reply:
refactored to `filteredResourceTypesSelector` into `allCollectionsFilteredResourceTypesSelector` which outputs an object with the first level key of `collectionId`s and the value is the previous output of `filteredResourceTypesSelector`. Also cleaned up any prior usage of `filteredResourceTypesSelector`